android/src/main/java/com/reactnative/unity/view/UnityViewManager.java

```diff
@@ -80,28 +79,11 @@ public void onHostDestroy() {
         }
     }
 
-    private void restoreUnityUserState() {
-        // restore the unity player state
-        if (UnityUtils.isUnityPaused()) {
-            Handler handler = new Handler();
-            handler.postDelayed(new Runnable() {
-                @Override
-                public void run() {
-                    if (UnityUtils.getPlayer() != null) {
-                        UnityUtils.getPlayer().pause();
-                    }
-                }
-            }, 300); //TODO: 300 is the right one?
-        }
-    }
-
     @Override
     public void onViewAttachedToWindow(View v) {
-        restoreUnityUserState();
```

Comment:
ale czy tutaj aby na pewno to nie będzie potrzebne? bo wtedy przy każdym dodaniu elementu do DOM będzie on ustawiany na resume (a mógł wcześniej być paused)

Reply:
Widok po dodaniu do drzewa nie wznawia się sam.

Kontrola nad pauzowanie/zatrzymywanie jest po stronie aplikacji poza sytuacją `onHostPause`/`onHostResume`. A to też chyba przeniosę do aplikacji, żeby już 100% było kontrolowane w jednym miejscu.

Reply:
Teraz występuje edge-case z timeoutem.

`setTimeout(() => UnityModule.pause(), …` nie zadziała poprawnie w momencie, gdy użytkownik robi pauzę.